### DISCUSSION TITLE
Windows firewall blocking notifications

### DISCUSSION THREAD

**Discussion by parsley72** [onvif/testspecs#59] [Q&A]
I've been trying to figure out why EVENT-2-1-17-v14.12 BASIC NOTIFICATION INTERFACE - NOTIFY fails for my device:
```
STEP 14 - Wait for notification
StepStart: 2023-07-13T02:13:15.9784445Z
No notification received!
STEP FAILED
```
I ran Wireshark and could see the notification message being sent, but the test failed as above. I eventually realised that although the TestTool.exe has a rule in the Windows Firewall for TCP and UDP packets from any port this doesn't work for packets sent from my device where the socket is opened from the device, like notifications.

Do you have a recommended way of setting up the firewall for this? Also can I suggest you add a note in the installation guide about this so others don't waste time trying to investigate it?

**Comment by madhurao68** [comment]
Hello,
In general, you set rules in the firewall to allow your applications to run. Since we do not know your current environment, we suggest you shut down the firewall and anti-virus application on your system and check if you encounter issues. You can narrow down to the problem by shutting down the anti-virus with the firewall on and vice versa to detect the real cause.
If things still do not work kindly let us know what rules you have set for the firewall. 
Thanks!

**Comment by parsley72** [comment]
I've already setup my firewall, I was just wondering if there were any recommendations. Also it took me several days to figure out what the problem was so it would be good to add this to the installation guide. It would also be good to change the ReadMe in the ONVIF_Device_TestTool installer that says:
> It would be highly appreciated if you provide your feedback (found bugs) to ONVIF workgroups

and gives the email address tswg_devicetest@developer.onvif.org because it doesn't work - it's taken me a while to find this forum.

Is the source code for the test tool available? I suspect that there's a way of setting up the network socket to receive notifications that Windows will handle correctly so this problem won't occur.

**Comment by madhurao68** [comment]
Hello,
You can contact the Testing Working Group chair through the Members Portal at https://members.onvif.org/working-groups-2/testing-working-group/. You will find options to send mail and subscribe to the Device and Client test tool mailing lists as well. 

>Is the source code for the test tool available? I suspect that there's a way of setting up the network socket to receive notifications that Windows will handle correctly so this problem won't occur.

The source code is not open source.

**Comment by parsley72** [comment]
I tried that and never got a response. The test tool tells you to report issues to an email address that doesn't work.
